core/backend/src/IModelSchemaLoader.ts

```diff
@@ -106,27 +105,18 @@ class IModelSchemaLocater implements ISchemaLocaterType {
    * @throws [IModelError]($common) if the schema exists, but cannot be loaded.
    */
   public getSchemaSync<T extends SchemaType>(schemaKey: typeof SchemaKey, _matchType: typeof SchemaMatchType, context?: typeof SchemaContext | undefined): T | undefined {
-    const schemaProps = this.getSchemaString(schemaKey.name);
-    if (!schemaProps)
-      return undefined;
-
-    context = context ? context : new SchemaContext();
-    return Schema.fromJsonSync(schemaProps, context) as T;
+    const schemaProps = this._iModel.nativeDb.getSchemaProps(schemaKey.name);
+    return Schema.fromJsonSync(schemaProps, context ?? new SchemaContext()) as T;
   }
 
-  /** Read schema data from the iModel as JSON string
-   * @param schemaName A string with the name of the schema to load.
-   * @returns A string with the JSON for the schema or `undefined` if the schema is not found.
-   * @throws [IModelError]($common) if the schema exists, but cannot be loaded.
+  /** Get a schema by [SchemaKey] asynchronously.
+   * @param schemaKey The [SchemaKey] that identifies the schema.
+   * * @param matchType The [SchemaMatchType] to used for comparing schema versions.
+   * @param context The [SchemaContext] used to facilitate schema location.
+   * @throws [Error] if the schema is not found or if the schema exists, but cannot be loaded.
    */
-  private getSchemaString(schemaName: string): string | undefined {
-    const val: IModelJsNative.ErrorStatusOrResult<any, any> = this._iModel.nativeDb.getSchema(schemaName);
-    if (undefined !== val.error) {
-      if (IModelStatus.NotFound === val.error.status) {
-        return undefined;
-      }
-      throw new IModelError(val.error.status, `reading schema=${schemaName}`);
-    }
-    return val.result;
+  public async getSchemaAsync<T extends SchemaType>(schemaKey: typeof SchemaKey, _matchType: typeof SchemaMatchType, context?: typeof SchemaContext | undefined): Promise<T | undefined> {
+    const schemaProps = await this._iModel.nativeDb.getSchemaPropsAsync(schemaKey.name);
+    return Schema.fromJsonSync(schemaProps, context ?? new SchemaContext()) as T;
```

Comment:
Just curious why this doesn't call the async `Schema.fromJson`.

Reply:
We still have some issues when multiple schemas that reference each other are being asynchronously loaded.